bootstrap/eks/controllers/eksconfig_controller_reconciler_test.go

```diff
@@ -125,6 +125,7 @@ func newEKSConfig(machine *clusterv1.Machine, name string) *bootstrapv1.EKSConfi
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: "default",
 			Name:      name,
+			UID:       types.UID(fmt.Sprintf("%s uid", name)),
```

Comment:
This feels wrong to me because I am impersonating the K8s API.

Reply:
hmmm i don't see that needing to be set in any of the CAPA tests. I'm looking at https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/controllers/awsmachine_controller_unit_test.go. i don't think it's that bad though. We are effectively impersonating the API by pretending the objects exist. 

🤔